src/app/[locale]/(main)/profile/[id]/profile/actions/update-user-profile.tsx

```diff
@@ -0,0 +1,46 @@
+import { getAPIServerURL } from "@/lib/utils"
+import { UpdateProfileForm } from "@/types/update-profile-form"
+
+type UpdateProfileSchema = {
+  id: string
+  token: string
+  values: UpdateProfileForm
+}
+
+export const updateUserProfile = async ({
+  id,
```

Comment:
all action server must be mark as "use server" and function have "action" as suffix (ex: getUserProfileAction())